delegator.py

```diff
@@ -107,14 +107,15 @@ def return_code(self):
     def std_in(self):
         return self.subprocess.stdin
 
-    def run(self, block=True):
+    def run(self, block=True, binary=False):
         """Runs the given command, with or without pexpect functionality enabled."""
         self.blocking = block
 
         # Use subprocess.
         if self.blocking:
-            s = subprocess.Popen(self._popen_args, **self._default_popen_kwargs)
-
+            popen_kwargs = self._default_popen_kwargs.copy()
+            popen_kwargs['universal_newlines'] = not binary
+            s = subprocess.Popen(self._popen_args, **popen_kwargs)
         # Otherwise, use pexpect.
         else:
             s = PopenSpawn(self._popen_args, **self._default_pexpect_kwargs)
```

Comment:
Hmm, so after some further testing, running your test case with `block=False, binary=True` still triggers the same bug we're trying to patch here. If we're going to add this, it needs to work well in both cases. Can you look into options for pexpect returning binary output?